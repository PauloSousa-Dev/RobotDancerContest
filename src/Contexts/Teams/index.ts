export { TeamsProvider, useTeamsContext } from "./Teams";
export type { Robot, Team } from "./Teams";
