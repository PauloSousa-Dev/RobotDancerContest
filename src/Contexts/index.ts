export { TeamsProvider, useTeamsContext } from "./Teams";
