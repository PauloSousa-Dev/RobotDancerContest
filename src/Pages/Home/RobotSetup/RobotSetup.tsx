import { useState } from "react";
import { TeamSetup } from "./TeamSetup";
import { DanceFloor } from "./DanceFloor";

const NUMBEROFTEAMS = 3;

type Robot = {
  id: number;
  name: string;
  powermove: string;
  experience: number;
  outOfOrder: boolean;
  avatar: string;
};

type Team = {
  id: number;
  name: string;
  robots: Robot[];
};

const isTeamsComplete = (teams: Team[]) =>
  teams.every((team) => team.robots.length === 5);

const RobotSetup = () => {
  const [teams, setTeams] = useState<Team[]>(
    Array(NUMBEROFTEAMS)
      .fill(null)
      .map((_, index) => ({
        id: index + 1,
        name: "",
        robots: [],
      }))
  );

  const isRobotsReadyToDance = isTeamsComplete(teams);

  if (isRobotsReadyToDance) {
    return <DanceFloor getTeams={teams} />;
  }

  return (
    <TeamSetup
      getTeams={teams}
      setTeam={(id: number, update: any) => {
        setTeams((teams) => {
          console.log(id, update);
          return teams.map((team) => {
            if (team.id === id) {
              return {
                ...team,
                ...update,
              };
            }
            return team;
          });
        });
      }}
    />
  );
};

export { RobotSetup };
