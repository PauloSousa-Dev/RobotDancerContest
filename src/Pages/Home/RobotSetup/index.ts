export { RobotSetup, NUMBEROFROBOTS } from "./RobotSetup";
