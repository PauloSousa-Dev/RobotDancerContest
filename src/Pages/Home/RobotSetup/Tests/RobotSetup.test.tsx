import React from "react";
import { render } from "@testing-library/react";
import { describe, expect, test } from "@jest/globals";
import { RobotSetup } from "../RobotSetup";
import { useTeamsContext } from "@/Contexts/__mocks__";

jest.mock("@/Contexts");

describe("Robot Setup component", () => {
  beforeEach(() => {
    jest.clearAllMocks(); // Clear previous mock calls and implementations
  });
  test("should render correctly", () => {
    const { container } = render(<RobotSetup />);
    expect(container).toMatchSnapshot();
  });

  test("should render correctly with setup complete", () => {
    (useTeamsContext as jest.Mock).mockReturnValueOnce({
      teams: [
        { id: 1, name: "Custom Team 1", robots: [1, 2, 3, 4, 5] },
        { id: 2, name: "Custom Team 2", robots: [1, 2, 3, 4, 5] },
      ],
      setTeam: jest.fn(),
      setRobots: jest.fn(),
      getRobotsById: jest.fn(),
      cleanWinners: jest.fn(),
      consts: { NUMBEROFTEAMS: 2, NUMBEROFROBOTS: 5 },
    });

    const { container } = render(<RobotSetup />);
    expect(container).toMatchSnapshot();
  });
});
