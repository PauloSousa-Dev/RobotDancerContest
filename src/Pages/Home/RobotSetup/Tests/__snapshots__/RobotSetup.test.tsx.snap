// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Robot Setup component should render correctly 1`] = `
<div>
  <div
    class="wrapper"
  >
    <h2>
      Set Up Teams
    </h2>
    <div
      class="inputWrapper"
    >
      <input
        placeholder="Team 1 Name"
        type="text"
        value="Team 1"
      />
      <input
        placeholder="Team 2 Name"
        type="text"
        value="Team 2"
      />
    </div>
    <button
      class="btn btn-primary btn-medium"
      type="button"
    >
      Fetch Robots
    </button>
  </div>
</div>
`;

exports[`Robot Setup component should render correctly with setup complete 1`] = `
<div>
  <h2>
    Disco Bar
  </h2>
  <div
    class="wrapper"
  >
    <div
      class="teamWrapper"
    >
      <div>
        <span
          class="teamName"
        >
          Team 
          Team 1
        </span>
      </div>
      <div>
        <span
          class="teamName"
        >
          Team 
          Team 2
        </span>
      </div>
    </div>
    <div
      class="buttonWrapper"
    >
      <button
        class="btn btn-primary btn-medium"
        type="button"
      >
        Start Contest
      </button>
    </div>
  </div>
</div>
`;
